pkg/registry/file/vulnsummarystorage.go

```diff
@@ -0,0 +1,240 @@
+package file
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+	"github.com/kubescape/storage/pkg/apis/softwarecomposition"
+	"github.com/spf13/afero"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/storage"
+)
+
+// VulnSummaryStorageImpl offers a common interface for object marshaling/unmarshaling operations and
+// hides all the storage-related operations behind it.
+type VulnSummaryStorageImpl struct {
+	appFs     afero.Fs
+	root      string
+	versioner storage.Versioner
+}
+
+var _ storage.Interface = &StorageImpl{}
+
+func NewVulnSummaryStorageImpl(appFs afero.Fs, root string) storage.Interface {
+	return &VulnSummaryStorageImpl{
+		appFs:     appFs,
+		root:      root,
+		versioner: storage.APIObjectVersioner{},
+	}
+}
+
+// Versioner Returns Versioner associated with this interface.
+func (s *VulnSummaryStorageImpl) Versioner() storage.Versioner {
+	return s.versioner
+}
+
+func (s *VulnSummaryStorageImpl) Create(ctx context.Context, key string, obj, out runtime.Object, _ uint64) error {
+	_, span := otel.Tracer("").Start(ctx, "VulnSummaryStorageImpl.Create")
+	span.SetAttributes(attribute.String("key", key))
+	defer span.End()
+	return storage.NewMethodNotImplementError(key, "")
+}
+
+func (s *VulnSummaryStorageImpl) Delete(ctx context.Context, key string, out runtime.Object, _ *storage.Preconditions, _ storage.ValidateObjectFunc, _ runtime.Object) error {
+	_, span := otel.Tracer("").Start(ctx, "VulnSummaryStorageImpl.Delete")
+	span.SetAttributes(attribute.String("key", key))
+	defer span.End()
+	return storage.NewMethodNotImplementError(key, "")
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) Watch(ctx context.Context, key string, _ storage.ListOptions) (watch.Interface, error) {
+	_, span := otel.Tracer("").Start(ctx, "VulnSummaryStorageImpl.Watch")
+	span.SetAttributes(attribute.String("key", key))
+	defer span.End()
+	return nil, storage.NewMethodNotImplementError(key, "")
+}
+
+// add uni test
```

Comment:
issue: this should not be in the code base

If we need to add unit test, the best course of action is to add it. If for some reason we cannot, we should add a `TODO(person_responsible)` with an explanation of what exactly we need to test and why we can’t do it now.

And having meaningful unit tests is mandatory before merging this PR.